### PR TITLE
Allow indexing of subfields

### PR DESCRIPTION
**Pull request by luca-rath**

### CODE REVIEW

File: Search/ObjectToDocumentConverter.php [hunk 1]
```diff
@@ -206,6 +206,32 @@ private function populateDocument($document, $object, $fieldMapping, $prefix = '
...
+                    /** @var Field $field */
+                    foreach ($value['fields'] as $field) {
+                        $field = clone $field;
+                        $field->setName($prefix . $fieldName . '#' . $field->getName());
+                        $document->addField($field);
+                    }
+                }
+
+                continue;
+            }
```
Comment (luca-rath) [inline Search/ObjectToDocumentConverter.php]: @wachterjohannes what do you think of that? This allows converters to return something like below

~~~php
[
    'value' => $id
    'fields' => [
        new Field('id', $id),
        new Field('name', $name),
    ],
]
~~~

**Comment by luca-rath** [comment]
ping @alexander-schranz @wachterjohannes

**Review by alexander-schranz — APPROVED** [review]

File: Search/ObjectToDocumentConverter.php (hunk 1 above)
Comment (wachterjohannes) [reply Search/ObjectToDocumentConverter.php]: 👍 

add this also to the docs

**Comment by wachterjohannes** [comment]
@luca-rath please add also some docs

**Review by wachterjohannes — APPROVED** [review]

**Comment by alexander-schranz** [comment]
@wachterjohannes anything missing for the release or why was no release tagged?

**Comment by wachterjohannes** [comment]
@alexander-schranz wanted to wait for the sulu PR - we can tag it now / @luca-rath will you do?

**Pull request merged**